httputil/request.go

```diff
@@ -172,14 +179,25 @@ func SetAuthType(req *http.Request, res *http.Response) {
 }
 
 func GetAuthType(res *http.Response) string {
-	auth := res.Header.Get("Www-Authenticate")
-	if len(auth) < 1 {
-		auth = res.Header.Get("Lfs-Authenticate")
+
+	authType := GetAuthTypeFromHeader(res, WwwAuthenticateHeader)
+	if authType == BasicAuthType {
+		// Let us check Lfs-Authenticate header to see if server supports NTML auth.
+		authType = GetAuthTypeFromHeader(res, LfsAuthenticateHeader)
 	}
 
-	if strings.HasPrefix(strings.ToLower(auth), "ntlm") {
-		return "ntlm"
+	return authType
+}
+
+func GetAuthTypeFromHeader(res *http.Response, headerName string) string {
+	authHeaders := res.Header[headerName]
+	for i := range authHeaders {
+		auth := authHeaders[i]
+
+		if strings.HasPrefix(strings.ToLower(auth), NtlmAuthType) {
+			return NtlmAuthType
+		}
 	}
 
-	return "basic"
+	return BasicAuthType
```

Comment:
It looks like this code only returns either the `Ntlm` or `Basic` auth types. I'm not super familiar with how this spec works: are there any other authentication types that we should support?